### PR TITLE
Thud compatability

### PR DESCRIPTION
**Pull request by rbdavison**
Slight modifications to allow building against the thud branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,10 @@ avoiding custom forks and binary solutions.
 This layer depends on:
 
 * URI: git://github.com/openembedded/openembedded-core
-  * branch: master
+  * branch: thud
   * revision: HEAD
 * URI: git://github.com/openembedded/bitbake
-  * branch: master
+  * branch: 1.42
```
Comment (alistair23) [inline README.md]: I don't test on non master branch so I don't think changing the README is a good idea

File: conf/layer.conf [hunk 2]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "pine64-layer"
 BBFILE_PATTERN_pine64-layer = "^${LAYERDIR}/"
 BBFILE_PRIORITY_pine64-layer = "5"
 
-LAYERSERIES_COMPAT_pine64-layer = "thud warrior"
+LAYERSERIES_COMPAT_pine64-layer = "thud"
```
Comment (alistair23) [inline conf/layer.conf]: Why remove warrior?

File: recipes-bsp/u-boot/u-boot_%.bbappend [hunk 3]
```diff
@@ -3,7 +3,7 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 SRC_URI += "file://boot.txt \
 	   "
 
-DEPENDS += "arm-trusted-firmware u-boot-tools-native"
+DEPENDS += "arm-trusted-firmware u-boot-mkimage-native"
```
Comment (alistair23) [inline recipes-bsp/u-boot/u-boot_%.bbappend]: Hmmm....

I don't want to change this to something that doesn't work with master, but I see the problem here. Is there any better way to do this?

**Comment by alistair23** [comment]
Thanks for the PR!

I added some comments, I don't think removing support for warrior/master is a good idea. Improving the thud support is probably worth doing.

The best option actually might be to create a thud branch, apply these patches to that and maintain a master and a thud branch. Would that work for you?

File: recipes-bsp/u-boot/u-boot_%.bbappend (hunk 3 above)
Comment (rbdavison) [reply recipes-bsp/u-boot/u-boot_%.bbappend]: When you build on the thud branch the u-boot-tools-native does not exist.

File: conf/layer.conf (hunk 2 above)
Comment (rbdavison) [reply conf/layer.conf]: because this change is for the thud branch only

File: README.md (hunk 1 above)
Comment (rbdavison) [reply README.md]: when building on the thud branch pointing to master doesn't work. However using 1.42 does.

**Comment by rbdavison** [comment]
The of the PR was to add the thud branch and update it so that it works.
If use the thud branch of bitbake and openembedded-core then the build fails.

In my case I want to include meta-mono and this fails with master and warrior and pointing to master is unstable.

**Comment by alistair23** [comment]
I have created a thud branch. Can you re-target this PR against that branch?

**Comment by alistair23** [comment]
Ah, I can re-target it. All done.

**Comment by rbdavison** [comment]
Awesome thank you, Thanks for the great work so far. Your work got me going and I'm grateful that I could assist in a tiny way.

**Comment by alistair23** [comment]
No worries. Always feel free to send a PR!

I'm glad people are using this layer. Let me know if it's missing anything.

**Pull request merged**